test/unit/test_diffmaps.py

```diff
@@ -0,0 +1,217 @@
+                index=pd.MultiIndex.from_tuples(miller_indices, names=["H", "K", "L"]),
+            ),
+            "SIGF_deriv": rs.DataSeries(
+                uncertainties_deriv,
+                dtype=np.float32,  # Ensure uncertainties are float32
+                index=pd.MultiIndex.from_tuples(miller_indices, names=["H", "K", "L"]),
+            ),
+            "PHI_nat": rs.DataSeries(
+                phases_native,
+                dtype=rs.PhaseDtype(),
+                index=pd.MultiIndex.from_tuples(miller_indices, names=["H", "K", "L"]),
+            ),
+            "PHI_deriv": rs.DataSeries(
+                phases_derivative,
+                dtype=rs.PhaseDtype(),
+                index=pd.MultiIndex.from_tuples(miller_indices, names=["H", "K", "L"]),
+            ),
+        }
+    )
+    return dataset
+
+
+# Smoke tests
+def test_compute_kweighted_deltafofo_smoke(sample_dataset: rs.DataSet) -> None:
+    result_dataset = compute_kweighted_deltafofo(
+        dataset=sample_dataset,
+        native_amplitudes="F_nat",
+        derivative_amplitudes="F_deriv",
+        native_phases="PHI_nat",
+        derivative_phases="PHI_deriv",
+        sigf_native="SIGF_nat",
+        sigf_deriv="SIGF_deriv",
+        kweight=1.0,
+    )
+    assert isinstance(result_dataset, rs.DataSet)
+    assert "DeltaFoFoKWeighted" in result_dataset.columns
+    assert "DeltaPhases" in result_dataset.columns
+
+
+def test_compute_deltafofo_smoke(sample_dataset: rs.DataSet) -> None:
+    result_dataset = compute_deltafofo(
+        dataset=sample_dataset,
+        native_amplitudes="F_nat",
+        derivative_amplitudes="F_deriv",
+        native_phases="PHI_nat",
+        derivative_phases="PHI_deriv",
+    )
+    assert isinstance(result_dataset, rs.DataSet)
+    assert "DeltaFoFo" in result_dataset.columns
+    assert "DeltaPhases" in result_dataset.columns
+
+
+# Test the correct output when DeltaFoFo is computed
+def test_compute_deltafofo_output(sample_dataset: rs.DataSet) -> None:
+    result_dataset = compute_deltafofo(
+        dataset=sample_dataset,
+        native_amplitudes="F_nat",
+        derivative_amplitudes="F_deriv",
+        native_phases="PHI_nat",
+        derivative_phases="PHI_deriv",
+    )
+
+    assert result_dataset is not None, "Function returned None when it shouldn't have."
+
+    # Test if the DeltaFoFo (amplitude difference) was correctly computed
+    delta_fofo = result_dataset["DeltaFoFo"].to_numpy()
+
+    # Compute expected DeltaFoFo using the updated approach (complex arithmetic)
+    native_complex = sample_dataset["F_nat"].to_numpy() * np.exp(
+        1j * np.deg2rad(sample_dataset["PHI_nat"].to_numpy())
+    )
+    derivative_complex = sample_dataset["F_deriv"].to_numpy() * np.exp(
+        1j * np.deg2rad(sample_dataset["PHI_deriv"].to_numpy())
+    )
+    expected_delta_complex = derivative_complex - native_complex
+    expected_delta_fofo = np.abs(expected_delta_complex)  # Amplitude differences
```

Comment:
Hmmm here you are just re-implementing the function.

That's a useful test if the test implementation is significantly simpler than the true implementation for some reason (e.g. speed), but I'd argue that's not the case here. The two are basically identical.

It would be better to test using some *results* that are, say, analytically verifiable. Ideas:
* one map minus itself should give zero
* map minus zero amplitude should give map, zero minus map gives negative map
* let map1 and map2 have amp 1 for all SFs. map1 phase = 0.0, map2=-90.0, then map1-map2 has phase 45.0